provider/provider.go

```diff
@@ -94,3 +94,17 @@ func getDefaultPath() string {
 		return "/usr/local/lib/summon"
 	}
 }
+
+// GetAllProviders creates slice of all file names in the default path
+func GetAllProviders(providerDir string) ([]string, error){
+	files, err := ioutil.ReadDir(providerDir)
+	if err != nil{
+		return make([]string, 0), err
+	}
+
+	names := make([]string, len(files))
+	for i, file := range files {
+        names[i] = file.Name()
+	}
+	return names, nil
+}
```

Comment:
This part has no unit tests around it. You have an integration one but no unit ones.